### PR TITLE
[workflows] fix syntax in codeql

### PR DESCRIPTION
**Pull request by eriknordmark**
Apparently the syntax was botched as part of fixing yamllint complaints., resulting in workflow failures of the form:
every step must define a `uses` or `run` key

This puts the '---' in the same place as for the other workflow files.

### CODE REVIEW

**Review by yash-zededa — APPROVED** [review]

**Comment by shjala** [comment]
@eriknordmark I missed your PR, I submitted #3379 for the same issue.

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -1,11 +1,11 @@
----
 # For most projects, this workflow file will not need changing; you simply need
 # to commit it to your repository.
 #
 # You may wish to alter this file to override the set of languages analyzed,
 # or to provide custom queries or build logic.
-name: "CodeQL"
 
+---
+name: "CodeQL"
```
Comment (milan-zededa) [inline .github/workflows/codeql.yml:8]: How is this change related to error `every step must define a uses or run key`? Or is this for yamllint?
Strange thing is that I do not see CodeQL action running for this PR.
But it seems that the patch from Shah fixed the issue and CodeQL action is green again.

**Comment by eriknordmark** [comment]
No longer needed.
